components/ide/gha-update-image/lib/jb-pin-version.ts

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) 2024 Gitpod GmbH. All rights reserved.
+// Licensed under the GNU Affero General Public License (AGPL).
+// See License.AGPL.txt in the project root for license information.
+
+import {
+    getLatestInstallerVersions,
+    pathToConfigmap,
+    readIDEConfigmapJson,
+    readWorkspaceYaml,
+    IWorkspaceYaml,
+} from "./common";
+
+const configmap = await readIDEConfigmapJson().then((d) => d.rawObj);
+
+const getIDEVersion = function (workspaceYaml: IWorkspaceYaml, ide: string) {
+    const url = workspaceYaml.defaultArgs[`${ide}DownloadUrl` as keyof typeof workspaceYaml.defaultArgs];
+    const str = url.split("-");
+    if (str.length < 2) {
+        return undefined;
+    }
+
+    return str.at(-1)!.replace(".tar.gz", "");
+};
+
+export const appendPinVersionsIntoIDEConfigMap = async () => {
+    const latestInstallerVersions = await getLatestInstallerVersions();
+    const workspaceYaml = await readWorkspaceYaml().then((d) => d.parsedObj);
```

Comment:
It can fix ENT-206 because we will read `WORKSPACE.yaml` again after previous file write of `jb-stable-version.ts` file